packages/next/src/client/components/navigation.ts

```diff
@@ -61,15 +61,15 @@ export function useSearchParams(): ReadonlyURLSearchParams {
   // In the case where this is `null`, the compat types added in
   // `next-env.d.ts` will add a new overload that changes the return type to
   // include `null`.
-  const readonlySearchParams = useMemo(() => {
+  const readonlySearchParams = useMemo((): ReadonlyURLSearchParams => {
     if (!searchParams) {
       // When the router is not ready in pages, we won't have the search params
       // available.
-      return null
+      return null!
     }
 
     return new ReadonlyURLSearchParams(searchParams)
-  }, [searchParams]) as ReadonlyURLSearchParams
+  }, [searchParams])
```

Comment:
drive-by type refactor to make it clear we only want the non-nullable case not an arbitrary type-case